src/core/ext/filters/http/client_authority_filter.cc

```diff
@@ -129,6 +129,16 @@ const grpc_channel_filter grpc_client_authority_filter = {
 
 static bool add_client_authority_filter(grpc_channel_stack_builder* builder,
                                         void* arg) {
+  const grpc_channel_args* channel_args =
+      grpc_channel_stack_builder_get_channel_arguments(builder);
+  const grpc_arg* a = grpc_channel_args_find(
+      channel_args, GRPC_ARG_DISABLE_CLIENT_AUTHORITY_FILTER);
+  if (a != nullptr) {
+    const bool disabled = grpc_channel_arg_get_bool(a, false);
+    if (disabled == true) {
```

Comment:
you can simply do `return is_client_authority_filter_disabled`

Reply:
No. If `is_client_authority_filter_disabled == false`, we want to continue with `grpc_channel_stack_builder_prepend_filter`.